client/src/views/Settings/Tabs/EditProfile/index.vue

```diff
@@ -145,47 +256,115 @@ export default {
   },
 
   created () {
-    this.$store.dispatch('uploadProfile')
+    this.$store.dispatch('profile/getMyProfile')
   },
 
   methods: {
-    createValue (val, done) {
-      if (val.length > 2) {
-        if (!suggestedInterests.includes(val)) {
-          done(val, 'add-unique')
-        }
+    uploadPhotoUrl (val) {
+      if (val.target.files[0].size > 2097152) {
+        this.$q.notify({
+          ...this.notifyParameters,
+          color: 'negative',
+          message: 'File is too big.'
+        })
+        this.$refs.avatar.val = ""
+        return
+      }
+      this.profile.avatar_url = val.target.files[0]
+      const reader = new FileReader()
+      reader.onload = () => {
+        this.photoUrl = reader.result
       }
+      reader.readAsDataURL(val.target.files[0])
     },
 
-    filterFn (val, update) {
-      update(() => {
-        if (val === '') {
-          this.filterOptions = suggestedInterests
-        } else {
-          const needle = val.toLowerCase()
-          this.filterOptions = suggestedInterests.filter(
-            v => v.toLowerCase().indexOf(needle) > -1
-          )
-        }
-      })
+    btnUploadPhotoUrl () {
+      this.$refs.photoUrlLabel.click()
     },
 
-    uploadAvatar (val) {
-      this.profile.avatar_url = val[0]
-      this.photoUrl = URL.createObjectURL(val[0])
+    deletePhotoUrl (val) {
+      this.photoUrl = ""
+      this.profile.avatar_url = ""
     },
 
     uploadBackgroundUrl (val) {
-      this.profile.user_background = val[0]
-      this.backgroundUrl = URL.createObjectURL(val[0])
+      if (val.target.files[0].size > 2097152) {
```

Comment:
You have almost identical method - uploadPhotoUrl, does it make sense to reuse it?